examples/mag/MAG_parser_v0.2.ipynb

```diff
@@ -0,0 +1,826 @@
+      "There are 53207043 authors and 12112687 of them have affiliations\n",
+      "There are 59310755 authors and 9551575 of them have affiliations\n",
+      "There are 55625205 authors and 5571860 of them have affiliations\n",
+      "There are 45392524 authors and 6678088 of them have affiliations\n",
+      "author ID data type: <class 'int'>\n"
+     ]
+    }
+   ],
+   "source": [
+    "full_authors = []\n",
+    "for i in range(5):\n",
+    "    f = open(f'mag_authors_{i}.txt', 'r')\n",
+    "    authors = []\n",
+    "    affs = []\n",
+    "    num_authors = 0\n",
+    "    for line in f.readlines():\n",
+    "        num_authors += 1\n",
+    "        author = json.loads(line)\n",
+    "        full_authors.append(author['id'])\n",
+    "        if 'last_known_aff_id' in author:\n",
+    "            authors.append(author['id'])\n",
+    "            affs.append(int(author['last_known_aff_id']))\n",
+    "            #assert author['last_known_aff_id'] in affs, f\"{author['last_known_aff_id']} does not exist\"\n",
+    "            #assert author['id'] not in author2aff\n",
+    "    df = pd.DataFrame({'author': authors, 'affiliation': affs})\n",
+    "    table = pa.Table.from_pandas(df)\n",
+    "    pq.write_table(table, f'mag_author2aff_{i}.parquet')\n",
+    "    print(f'There are {num_authors} authors and {len(authors)} of them have affiliations')\n",
+    "    \n",
+    "df = pd.DataFrame({'id': full_authors})\n",
+    "print('author ID data type:', type(full_authors[0]))\n",
+    "table = pa.Table.from_pandas(df)\n",
+    "pq.write_table(table, 'mag_author_id.parquet')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def parse_paper_file(i):\n",
+    "    f = open(f'mag_papers_{i}.txt', 'r')\n",
+    "    num_paper_venue = 0\n",
+    "    venue_ids = set()\n",
+    "    fos_set = set()\n",
+    "\n",
+    "    paper_ids = []\n",
+    "    titles = []\n",
+    "    years = []\n",
+    "    paper2author = ([], [], [])\n",
+    "    paper2venue = ([], [])\n",
+    "    paper2fos = ([], [], [])\n",
+    "    paper2paper = ([], [])\n",
+    "    for line in f.readlines():\n",
+    "        paper = json.loads(line)\n",
+    "        if 'id' not in paper or 'title' not in paper or 'year' not in paper or 'authors' not in paper or len(paper['authors']) == 0:\n",
+    "            continue\n",
+    "        paper_ids.append(paper['id'])\n",
+    "        titles.append(paper['title'])\n",
+    "        years.append(paper['year'])\n",
+    "        for order, author in enumerate(paper['authors']):\n",
+    "            paper2author[0].append(paper['id'])\n",
+    "            paper2author[1].append(author['id'])\n",
+    "            paper2author[2].append(order)\n",
+    "        #if len(paper['authors']) == 0:\n",
+    "        #    print(paper)\n",
+    "        if 'venue' in paper:\n",
+    "            num_paper_venue += 1\n",
```

Comment:
Should we move L179 under `if 'id' in vanue:`?

Reply:
If id not in venue, we do not record it.

Reply:
i'm not sure if it's necessary.